### ISSUE TITLE
xish.xmlstream.XmlStreamFactory should not inherit from protocol.ReconnectingClientFactory

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3492]
|[<img alt="philmayers's avatar" src="https://avatars.githubusercontent.com/u/739238?s=50" width="50" height="50">](https://github.com/philmayers)| @philmayers reported|
|-|-|
|Trac ID|trac#3492|
|Type|enhancement|
|Created|2008-10-24 14:35:20Z|

As above. AFACIT it's trivial to mix-in p.ReconnectingBlah later on if needed. Explicit is better than implicit I think.

<details><summary>Searchable metadata</summary>

```
trac-id__3492 3492
type__enhancement enhancement
reporter__philmayers philmayers
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__words words
keywords__ 
time__1224858920000000 1224858920000000
changetime__1224954647000000 1224954647000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set status to `closed`|
|-|-|

`XmlStreamFactory` was included in Twisted 8.1 as a subclass of `ReconnectingClientFactory`.  Perhaps bad, but changing it now would break any code relying on this.  So while I agree that there seems to be no good reason for `XmlStreamFactory` to have subclassed RCF, it's too late to change now.

Perhaps there is some other change in this area which we could make to improve the situation, though.

I suggest talking with ralphm about factories and coming up with a new ticket.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
